lib/scala/pkg/src/main/java/org/enso/pkg/archive/CloseArchiveEntryException.java

```diff
@@ -0,0 +1,11 @@
+package org.enso.pkg.archive;
+
+import java.io.IOException;
+
+/** Error when trying to close an enso-project archive entry. */
+public final class CloseArchiveEntryException extends EnsoProjectArchiveException {
+
+  public CloseArchiveEntryException(IOException cause) {
```

Comment:
If the class (and other similar cases) is only constructed in the same package, then you may want to have _package private_ constructors.